src/feedback.rs

```diff
@@ -0,0 +1,235 @@
+#![allow(missing_docs)]
+
+use fxhash::FxHashSet;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+use std::default::Default;
+use std::hash::Hash;
+
+use crate::VirtAddr;
+
+pub type TaggedFeedback = BTreeMap<u64, FxHashSet<u64>>;
+pub type ScratchFeedback = BTreeSet<VirtAddr>;
+pub type MaxFeedback = BTreeMap<u64, u64>;
+pub type CustomFeedback = FxHashSet<u64>;
+
+#[cfg(feature = "redqueen")]
+use x86_64::registers::rflags::RFlags;
+#[cfg(feature = "redqueen")]
+pub type RedqueenFeedback = BTreeSet<(VirtAddr, RFlags)>;
+
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum FeedbackLog {
+    VAddr(VirtAddr),
+
+    #[cfg(feature = "custom_feedback")]
+    Custom(u64),
+    #[cfg(feature = "custom_feedback")]
+    CustomTagged((u64, u64)),
+    #[cfg(feature = "custom_feedback")]
+    CustomMax((u64, u64)),
+
+    #[cfg(feature = "redqueen")]
+    #[serde(skip)]
+    Redqueen((VirtAddr, RFlags)),
```

Comment:
@corydu I copied this from the old redqueen code, but this actually doesn't compile because `RFlags` is not `Serializable`. Not sure if it is really needed though.

Reply:
Yeah we could just use a u64 for this and not the actual `RFlags` itself.